File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
       int minprice = prices[0];
        int profit=0;
        for(int i=1;i<prices.size();i++){
            if(prices[i]<minprice)
            {
                minprice = prices[i];
            }
            else{
                profit = max(prices[i]-minprice,profit);
            }
        }
        return profit;
    }
};

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
        int profit=0;
        int buyed=prices[0];
        for(int i=1;i<prices.length;i++){
            if(buyed < prices[i]){
                profit = Math.max(profit,prices[i]-buyed);
            }
            else{
                buyed = prices[i];
            }
        }
        return profit;
    }
}